src/Datadog.Trace/Sampling/RuleBasedSampler.cs

```diff
@@ -74,7 +74,7 @@ public void RegisterRule(ISamplingRule rule)
 
         private SamplingPriority GetSamplingPriority(Span span, float rate, bool agentSampling)
         {
-            var sample = ((span.TraceId * KnuthFactor) % TracerConstants.MaxTraceId) <= (rate * TracerConstants.MaxTraceId);
+            var sample = ((span.TraceId * KnuthFactor) % TracerConstants.MaxTraceId) < (rate * TracerConstants.MaxTraceId);
```

Comment:
If `rate` is zero, can we shortcut and return `AutoReject`?

Reply:
Similarly if `rate` is 1.0 (which is probably the most common value).

Reply:
Does this change have any effect? If `rate = 1.0`, then the right side of this comparison will resolve to `TracerConstants.MaxTraceId`, but left side can never be equal to `TracerConstants.MaxTraceId` because of the modulus operator.